crates/provider/src/shared/types.rs

```diff
@@ -0,0 +1,185 @@
+use std::{
+    collections::HashMap, fs::File, io::Write, os::unix::process::ExitStatusExt, path::PathBuf,
+    process::ExitStatus, vec,
+};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum ZombieRole {
+    Temp,
+    Node,
+    BootNode,
+    Collator,
+    CumulusCollator,
+    Authority,
+    FullNode,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum PortName {
+    Prometheus,
+    Rpc,
+    RpcWs,
+    P2P,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+enum ImagePullPolicy {
+    IfNotPresent,
+    Never,
+    Always,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct FileMap {
+    local_file_path:  PathBuf,
+    remote_file_path: PathBuf,
+    unique:           bool,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct RunCommandResponse {
+    pub exit_code: ExitStatus,
+    pub std_out:   Vec<u8>,
+    pub std_err:   Option<Vec<u8>>,
+}
+
+impl RunCommandResponse {
+    pub fn default() -> Self {
+        Self {
+            exit_code: ExitStatus::from_raw(0),
+            std_out:   vec![],
+            std_err:   None,
+        }
+    }
+}
+
+#[derive(Debug, Default, Clone, PartialEq)]
+pub struct NativeRunCommandOptions {
+    pub allow_fail: bool,
```

Comment:
Following [rust convention](https://doc.rust-lang.org/1.0.0/style/style/naming/README.html#predicates), we should prefix boolean with ```is_```:

```suggestion
    pub is_failure_allowed: bool,
```